### PR TITLE
Re-add -Dopenssl to zap build

### LINKED ISSUE #62: Re-add the -Dopenssl option to zap build
Hey, I'd like to bring this up for discussion at least.

I didn't even really know any of this myself until fiddling with it last night, but users of zap _could_ pass the -Dopenssl flag to the zap build from their own build.zig via the following:

```zig
const zap = b.dependency("zap", .{
    .target = target,
    .optimize = optimize,
    .openssl = true,
});
```

I find this approach much more simple than having to pass in the environment variable each time I want to build my own project (or by setting it some other way and having that be a pre-setup step to building my project).

I'd at least like the -Dopenssl flag still so I can continue to do this, but only because I find this solution rather elegant.

### PR DESCRIPTION
**Pull request by Vemahk**
Also includes a small modification to the readme for instructions how to pass it through from any project that depends on zap.

Would resolve #62 if accepted.

Lemme know what ya think.

### CODE REVIEW

File: build.zig
Comment (renerocksai) [inline build.zig:10]: That use of orelse is actually very clever!

File: README.md
Comment (renerocksai) [inline README.md:276]: I am not entirely sure if it shouldn't read `.openssl = false;   // set to true for TLS ...`?

**Review by renerocksai — APPROVED** [review]
I think I like my suggestion for the readme better.

**Comment by Vemahk** [comment]
Oh dang, you're pretty fast on that.  Noticed I left the comment behind without reflecting the change.  Get some sleep, my dude, this can wait 😅

File: README.md
Comment (Vemahk) [reply README.md:276]: Yeah, that works. I'll switch that around.

**Comment by renerocksai** [comment]
Okokok. G'night 💤

**Comment by Vemahk** [comment]
Okay, I'm done...  I at least know how git amend works, now 🤔.

Lemme know if you see any other docs that might need to be updated.

**Comment by renerocksai** [comment]
I'll merge right after the CI checks have been completed, then add one more commit to the README, and replay the whole shebang on the zig-0.12.0 branch. Will let you know when ready.
- [x] CI checks passed
- [x] PR merged
- [ ] add 1 more commit to README.md (minor formatting stuff)
- [ ] replay all commits on zig-0.12.0 branch

**Pull request merged**
